### PR TITLE
Improved Car service

### PR DESCRIPTION
**Pull request by ArtemReizer**

### CODE REVIEW

File: src/main/java/com/jvavateam/carsharingapp/service/impl/CarServiceImpl.java [hunk 1]
```diff
@@ -48,9 +50,27 @@ public void deleteById(Long id) {
...
         Car updatedCar = carRepository.findById(car.getId()).orElseThrow(
-                () -> new RuntimeException("Can't find car by id: " + car.getId())
+                () -> new EntityNotFoundException("Can't find car by id: " + car.getId())
         );
         updatedCar.setInventory(car.getInventory());
         carRepository.save(updatedCar);
     }
+
+    @Override
+    public Car updateCar(Car car) {
```
Comment (Dimagaa) [inline src/main/java/com/jvavateam/carsharingapp/service/impl/CarServiceImpl.java]:
```suggestion
    public Car updateCar car) {
```

File: src/main/java/com/jvavateam/carsharingapp/service/impl/CarServiceImpl.java [hunk 2]
```diff
@@ -48,9 +50,27 @@ public void deleteById(Long id) {
...
+        Car updatedCar = new Car();
+        updatedCar.setBrand(car.getBrand());
+        updatedCar.setModel(car.getModel());
+        updatedCar.setType(car.getType());
+        updatedCar.setInventory(car.getInventory());
+        updatedCar.setDailyFee(car.getDailyFee());
+        return updatedCar;
```
Comment (Dimagaa) [inline src/main/java/com/jvavateam/carsharingapp/service/impl/CarServiceImpl.java]: return carRepository.save(car)

**Review by Dimagaa — CHANGES_REQUESTED** [review]

File: src/main/java/com/jvavateam/carsharingapp/service/impl/CarServiceImpl.java [hunk 3]
```diff
@@ -48,9 +61,16 @@ public void deleteById(Long id) {
...
         Car updatedCar = carRepository.findById(car.getId()).orElseThrow(
-                () -> new RuntimeException("Can't find car by id: " + car.getId())
+                () -> new EntityNotFoundException("Can't find car by id: " + car.getId())
         );
         updatedCar.setInventory(car.getInventory());
         carRepository.save(updatedCar);
     }
+
+    @Override
+    public Car findCarById(Long id) {
```
Comment (holsoni) [inline src/main/java/com/jvavateam/carsharingapp/service/impl/CarServiceImpl.java]: findById
```suggestion
    public Car findById(Long id) {
```

File: src/main/java/com/jvavateam/carsharingapp/service/impl/CarServiceImpl.java [hunk 4]
```diff
@@ -48,9 +61,16 @@ public void deleteById(Long id) {
     @Override
     public void updateCarInventory(Car car) {
```
Comment (holsoni) [inline src/main/java/com/jvavateam/carsharingapp/service/impl/CarServiceImpl.java]:
```suggestion
    public void updateInventory(Car car) {
```

**Review by holsoni — CHANGES_REQUESTED** [review]
I think that it's better not ot use 'car' word in method namings

File: src/main/java/com/jvavateam/carsharingapp/service/impl/CarServiceImpl.java [hunk 5]
```diff
@@ -23,14 +24,26 @@ public CarDtoResponse create(CarDtoRequest carDto) {
...
+    }
+
+    @Override
+    public Car update(Car car) {
+        Car updatedCar = new Car();
+        updatedCar.setBrand(car.getBrand());
+        updatedCar.setModel(car.getModel());
+        updatedCar.setType(car.getType());
+        updatedCar.setInventory(car.getInventory());
+        updatedCar.setDailyFee(car.getDailyFee());
```
Comment (Dimagaa) [inline src/main/java/com/jvavateam/carsharingapp/service/impl/CarServiceImpl.java]: Why do you need to create a new car?

**Review by Dimagaa — APPROVED** [review]
Well done!

**Review by Dimagaa — APPROVED** [review]

**Review by holsoni — APPROVED** [review]

**Pull request merged**